### DISCUSSION TITLE
Storage deposits for every call request

### DISCUSSION THREAD

**Discussion by zavodil** [near/near-sdk-js#104] [Q&A] [answered]
How to make call methods without storage deposit on every request? This storage deposit is fine for near-cli usage, but it breaks UX with web app usage because of redirects to near-wallet. Can we implement methods to storage deposit once “in advance” and then use it for storage for specific user?

**Comment by hiraqdev** [comment]
It's interesting with your case @zavodil , i'm just begin to interest to develop smart contract on top of NEAR blockchain using `near-sdk-js` and  i'm just starting to learn about [storage staking](https://nomicon.io/Standards/StorageManagement#nep-145) standard from NEAR

i'm just curious that why you need to deposit storage for every call request from web app? is it expected behaviors? because when i'm learning standard storage staking (NEP 145) i though that we can measure contract's storage minimal balance via `StorageBalanceBounds` , so the web app can query this balance's value before their user call some contract's method, and then suggest their user to deposit at least to achieve storage's minimal balance

**Comment by volovyks** [comment, marked as answer]
@zavodil used an old (enclaved) design of the NEAR SDK JS. With the current SDK design, we no longer have such a problem.
